Add metadata import section

pwalczysko opened the pull request:
Adding metadata import section.

Possibly the order could be reversed, and possibly a new section showing how to create a quick and easy, simple bulkmap config could come in. 

cc @francesw @manics @jburel

pwalczysko committed:
Add metadata import section

mtbc commented on docs/metadata.rst:
```diff
@@ -0,0 +1,98 @@
+Import metadata using the Command Line Interface (CLI)
+=======================================================
```
```suggestion
======================================================
```

mtbc commented on docs/metadata.rst:
```diff
@@ -0,0 +1,98 @@
+
+We will show:
+
+-  How to import metadata from local CSV file in a bulk manner and turn them into OMERO.tables on images using CLI
```
```suggestion
- How to import metadata from local CSV file in a bulk manner and turn them into OMERO.tables on images using CLI
```

mtbc commented on docs/metadata.rst:
```diff
@@ -0,0 +1,98 @@
+
+#.  Go to your browser and OMERO.web, select the images in the Project you targeted and verify that they have now new Key-Value pair annotations displayed in the right-hand pane.
+
+#.  Still in OMERO.web, create a new Dataset and copy into it 4 images, preferably images which have nor OMERO.tables on them neither any Key-Value pairs attached. Note the names of the images you are copying in.
```
```suggestion
#.  Still in OMERO.web, create a new Dataset and copy into it 4 images, preferably images which have neither OMERO.tables on them nor any Key-Value pairs attached. Note the names of the images you are copying in.
```

manics commented on docs/metadata.rst:
```diff
@@ -0,0 +1,98 @@
+
+#.  Go back to your terminal. Download the idr0021-experimentA-bulkmap-config.yml file https://github.com/IDR/idr0021-lawo-pericentriolarmaterial/blob/9479af85f19487f215e3dfdd31a1b587370ed3cf/experimentA/idr0021-experimentA-bulkmap-config.yml.
+
+#.  Open the downloaded file in a text editor and delete the ``Advanced options...`` section. Save the file and run::
```
I think it would be clearer if you bundled a modified copy of that file in this repository instead of telling people to download an external file and modify it.

joshmoore replied:
It's an interesting question. Benefits of using the original would include:
 - no need to keep changes synchronized
 - teaching users that there's a wealth of examples under GitHub.com/idr
Benefits of a copy would include:
 - clarity as you say
 - Ability to perhaps simplify the file

cc: @sbesson

manics replied:
Do you anticipate needing to modify the released idr0021 study?

pwalczysko replied:
@joshmoore @manics Please note https://github.com/ome/omero-metadata/issues/44 - this is the main reason why not to use the idr studies workflows as they are in idr and simply transfer the workflow on outreach.

manics commented on docs/metadata.rst:
```diff
@@ -0,0 +1,98 @@
+
+#.  Download the csv from https://github.com/IDR/idr0021-lawo-pericentriolarmaterial/blob/9479af85f19487f215e3dfdd31a1b587370ed3cf/experimentA/idr0021-experimentA-annotation.csv 
+
+#.  The variable ``$ID​`` below is the ID of the ​Project, in this example case it is the Project containing the idr0021 study. To add annotations from a local csv file to the images in the said Project in the form of OMERO.tables, run::
```
This assumes the user has imported `idr0021` somewhere, but I can't see any mention of it in https://omero-guides.readthedocs.io/en/latest/upload/docs/import.html

pwalczysko replied:
No, but the data are mentioned in this very guide, under Resources. The thinking is - if you are in our workshop, you will have the data there ready for you, so no worries. If you are outwith the workshop, you will probably not want to be importing idr0021 on your server. just to test the workflows.  You will want to have the files such as bulkmap and tsv and compare the patterns there with the data via navigating to idr.

manics commented on docs/metadata.rst:
```diff
@@ -0,0 +1,98 @@
+
+#.  Still in OMERO.web, create a new Dataset and copy into it 4 images, preferably images which have nor OMERO.tables on them neither any Key-Value pairs attached. Note the names of the images you are copying in.
+
+#.  Go to the https://pypi.org/project/omero-metadata/ and find the section named ``populate``. Copy the ``project.csv`` from that section and paste it locally, either into your text editor, or into Microsoft Excel. Edit the names of the images in the first column to match the names of the images you copied into your Dataset in the previous step. Also, edit the name of the Dataset in the second c ...
```
Can you bundle this file in this repository so it can be downloaded, instead of telling users to copy and paste?

pwalczysko replied:
fixed in https://github.com/ome/omero-guide-upload/pull/6/commits/4554098849c4eb991f81408df8dfce8d3cfcc0ae

manics reviewed (commented):
It's not clear to me who this guide is targetted at:
- People using the outreach server during a workshop given by us?
- People using their own OMERO server?

pwalczysko commented:
@manics 



It's not clear to me who this guide is targetted at:

    People using the outreach server during a workshop given by us?
    People using their own OMERO server?


Both. We need simply both. And we do not have the margin to produce two parallel lines.
If you could specify the main unclear points (for potential users who are not within our workshop) ?

manics commented:
I think you need to tell people how to obtain and import `idr0021`, otherwise they can't follow the guide on their own server. Alternatively you could write your own `bulkmap-config.yml` for a smaller dataset?

For example, https://omero-guides.readthedocs.io/en/latest/introduction/docs/annotate.html says
> The data used are from the siRNAi-HeLa folder available at https://downloads.openmicroscopy.org/images/DV/siRNAi-HeLa/

Perhaps you could annotate those images instead?

joshmoore commented on docs/metadata.rst:
```diff
@@ -0,0 +1,98 @@
+
+-  Documentation:
+
+   -  https://docs.openmicroscopy.org/latest/omero/users/cli/installation.html
```
Perhaps use the same omero/latest or latest/omero everywhere. (see next line)

pwalczysko replied:
Fixed in https://github.com/ome/omero-guide-upload/pull/6/commits/321c12605df82d3ab121873b8eb4ca546a87ce74

joshmoore commented on docs/metadata.rst:
```diff
@@ -0,0 +1,98 @@
+    
+    $ omero metadata populate --report --batch 1000 --file local/path/to/idr0021-experimentA-annotation.csv Project:$ID
+
+#.  Open your browser and login to the OMERO.web. Navigate to the Project you just worked with, expand the "Attachments" harmonica in the right-hand pane and verify that a new attachment is on that Project named ``bulk_annotations``.
```
I imagine some of these non-CLI steps might benefit from screenshots.

pwalczysko committed:
Add simple annotation csv

pwalczysko committed:
Delete the "undefined" items, empty cells will be ignored

manics committed:
Add example simple-annotation-bulkmap-config.yml

This corresponds to simple-annotation.csv

For example, run:

omero metadata populate --context bulkmap --cfg simple-annotation-bulkmap-config.yml Dataset:2066

To delete the map-annotations added by the above config file run:

omero metadata populate --context deletemap --cfg simple-annotation-bulkmap-config.yml Dataset:2066

manics commented:
https://github.com/pwalczysko/omero-guide-upload/pull/1 adds a `simple-annotation-bulkmap-config.yml` file corresponding to `simple-annotation.csv`

pwalczysko committed:
Merge pull request #1 from manics/pr6

Add example simple-annotation-bulkmap-config.yml

pwalczysko committed:
Update docs/metadata.rst

Co-authored-by: Mark Carroll <m.t.b.carroll@dundee.ac.uk>

pwalczysko committed:
Update docs/metadata.rst

Co-authored-by: Mark Carroll <m.t.b.carroll@dundee.ac.uk>

pwalczysko committed:
Update docs/metadata.rst

Co-authored-by: Mark Carroll <m.t.b.carroll@dundee.ac.uk>

pwalczysko commented:
@manics thank you for the ``simple-bulkmap-config.yml``

After reading the description in the commit and the comments in the file itself, I have two questions.

Can we have the KVPs listed under the "Default namespace" organized in the "Others" header in the webclient ? What governs this ?  @francesw was meaning that this might be the indentations of the blocks such as https://github.com/ome/omero-guide-upload/pull/6/commits/d9acb71c5f3a1406f7454820e475095795407067#diff-e2d81d8cdbf9ac42c679d54ca6b0af06R51 (in her example these blocks had indentations, in yours https://github.com/ome/omero-guide-upload/pull/6/commits/d9acb71c5f3a1406f7454820e475095795407067#diff-e2d81d8cdbf9ac42c679d54ca6b0af06R51 they are unindented. Is that the crucial point which decides about the "Others" header in webclient later ?

We do not have all the mapr categories defined in the outreach servers, and, somehow, the mapr on these servers does not seem to work for these newly added KVPs using the ``simple...`` csv and bulkmap files. Only the "old" annotated idr0021 seem to be handled correctly by mapr on outreach.

This (and also the fact that mapr is complex, needs an extra install, we do not have a walkthrough for it, etc.) leads me to another question:

Can I have a nice header like Phenotypes for my KVP group, but not have a mapr category defined ? (as you see, I would like to offer a step 1. no-mapr-yet and step 2. mapr-if-you-want approach).

pwalczysko commented:
@manics Is this jinja magic https://github.com/ome/omero-guide-upload/pull/6/commits/d9acb71c5f3a1406f7454820e475095795407067#diff-e2d81d8cdbf9ac42c679d54ca6b0af06R107 logic encoded in the omero metadata plugin ? This is a nice feature.

pwalczysko committed:
Add downloadable csv cf. Simons comment

pwalczysko committed:
Unify links order cf. Josh comment

pwalczysko committed:
Add screenshots, improve text

pwalczysko committed:
Add the simple file for siRNA-HeLa

pwalczysko commented:
Addressed the comments.  Now we have two options, one using the IDR data, one using the siRNAi-HeLa data, which are easier to get.

jburel commented on docs/metadata.rst:
```diff
@@ -0,0 +1,165 @@
+
+-  Data: example images from
+
+   -  IDR data http://idr.openmicroscopy.org/webclient/?show=project-51
```
https

jburel commented:
maybe create a folder with the ``csv, yml`` files and not mix them with the ``.rst`` doc file

jburel commented on docs/metadata.rst:
```diff
@@ -0,0 +1,165 @@
+    
+    $ pip install omero-metadata
+
+**Step-by-step:**
```
I moved away from making it bold i.e. remove ``**...**``

jburel commented on docs/metadata.rst:
```diff
@@ -0,0 +1,165 @@
+
+#.  On your local machine, open a terminal
+
+#.  Activate the virtual environment where ``omero-py`` is installed or add it to ``PATH`` e.g.::
```
The question that a user could ask. is it the same virtual env than the one above?
Do I need 2 venv?

jburel commented on docs/metadata.rst:
```diff
@@ -0,0 +1,165 @@
+
+    $ export PATH=/opt/omero/server/venv3/bin:$PATH
+
+#.  Download the csv from https://github.com/IDR/idr0021-lawo-pericentriolarmaterial/blob/9479af85f19487f215e3dfdd31a1b587370ed3cf/experimentA/idr0021-experimentA-annotation.csv if you have access to the idr0021 data http://idr.openmicroscopy.org/webclient/?show=project-51 in your OMERO.server. Alternatively, download :download:`simple-annotation.csv <simple-annotation.csv>`, which will allow you ...
```
CSV

pwalczysko replied:
Not sure what you mean - what is here to correct ?

jburel replied:
use capital

pwalczysko replied:
@jburel I think I fixed it , you mean the capital CSV - I fixed for all CSV in the doc - just could not find the CSV in that particular section

jburel commented on docs/metadata.rst:
```diff
@@ -0,0 +1,165 @@
+
+    $ omero metadata populate --context bulkmap --cfg local/path/to/simple-annotation-bulkmap-config.yml --batch 100 Dataset:$ID
+
+#.  Go to your browser and OMERO.web, select the images in the Project or Dataset you targeted and verify that they have now new Key-Value pair annotations displayed in the right-hand pane.
```
sometimes it is ``Key-Value pair annotations`` sometimes ``Key-Value pairs``

jburel commented on docs/metadata.rst:
```diff
@@ -0,0 +1,165 @@
+
+    |image3a|
+
+#.  Still in OMERO.web, create a new Dataset and copy into it 4 images, preferably images which have neither OMERO.tables on them nor any Key-Value pairs attached. Note the names of the images you are copying in.
```
four

jburel commented on docs/metadata.rst:
```diff
@@ -0,0 +1,165 @@
+
+    |image4|
+
+#.  Go to the https://pypi.org/project/omero-metadata/ and find the section named ``populate``. Study the ``project.csv``. You can either take the ``project.csv`` file from there, or more conveniently, you can download directly its copy :download:`four-images.csv <four-images.csv>`. Open the csv in Excel and edit the names of the images in the first column to match the names of the images you cop ...
```
CSV

jburel commented on docs/metadata.rst:
```diff
@@ -0,0 +1,165 @@
+
+#.  Go to the https://pypi.org/project/omero-metadata/ and find the section named ``populate``. Study the ``project.csv``. You can either take the ``project.csv`` file from there, or more conveniently, you can download directly its copy :download:`four-images.csv <four-images.csv>`. Open the csv in Excel and edit the names of the images in the first column to match the names of the images you cop ...
+
+#.  In your OMERO.web, upload the csv you just saved and attach it onto the Dataset you created previously.
```
same

pwalczysko committed:
Fix text cf. Jmaries comments

jburel committed:
review header

jburel committed:
relocate files

jburel committed:
update links

jburel committed:
review link display

manics commented on scripts/simple-annotation-bulkmap-config.yml:
```diff
@@ -0,0 +1,135 @@
+        # "clientvalue" can be used to override the value of the map-
+        # annotation key-value. The "value" is substituted using
+        # Jinja2 templates.
+        clientvalue: http://www.ensembl.org/id/{{ value|urlencode }}
```
```suggestion
        clientvalue: https://www.ensembl.org/id/{{ value|urlencode }}
```

manics reviewed (commented):
Looks fine to me

pwalczysko committed:
Update scripts/simple-annotation-bulkmap-config.yml

Co-authored-by: Simon Li <orpheus+devel@gmail.com>
